lib/email.rb

```diff
@@ -0,0 +1,26 @@
+class Email
+  def initialize(user)
+    @address = user.email
+    @name = user.email.split(/@/).first
+  end
+end
+
+class WelcomeEmail < Email
```

Comment:
I'm not sure about using class inheritance to distinguish between types of email. Personally I would probably opt to make `subject` and `body` a dependency of the `Email` class, and define these as constants elsewhere.